Frontend/isbit/src/components/dataset-selection/DataSelection.vue

```diff
@@ -1,121 +1,283 @@
 <template>
   <div class="container">
-    <h1>Dataset Information</h1>
+    <h1 class="header">Dataset Information</h1>
     <div v-if="error" class="error">{{ error }}</div>
-    <!-- Error handling -->
-    <div class="card" v-for="(item, index) in dataList" :key="index">
+    <div class="card" v-for="(item, index) in combinedDataList" :key="index">
       <div class="card-content">
-        <h2>{{ item.dataset }}</h2>
+        <h2 class="card-title">{{ item.dataset }}</h2>
         <div class="card-details">
           <div class="label-group">
-            <p><strong>Uppgift:</strong> {{ item.assignment }}</p>
-            <p><strong>Typ av data:</strong> {{ item.datatype }}</p>
+            <p v-if="isDataAvailable(item.dataset)" class="label">
+              <i class="icon assignment-icon"></i>
+              <strong>Uppgift:</strong> {{ item.assignment }}
+            </p>
+            <p v-if="isDataAvailable(item.dataset)" class="label">
+              <i class="icon datatype-icon"></i>
+              <strong>Typ av data:</strong> {{ item.datatype }}
+            </p>
+            <p class="availability">
+              <strong>Tillgänglighet:</strong>
+              {{ isDataAvailable(item.dataset) ? 'Tillgänglig' : 'Ej tillgänglig' }}
+            </p>
+          </div>
+
+          <div class="button-container">
+            <button 
+              v-if="isDataAvailable(item.dataset)" 
+              @click="redirectToDetails(item.dataset)"
+              class="btn primary-btn"
+            >
+              Märk Upp Data
```

Comment:
Should probably write the last two words without any capital letters: Märk upp data, especially since you only use a capital letter for the first word in the "Ta bort dataset" button.

Reply:
yes, fixed this. I'm used to the English way of writing it.